app/models/subject.rb

```diff
@@ -11,7 +13,8 @@ class Subject < ActiveRecord::Base
   has_many :collections, through: :collections_subjects
   has_many :subject_sets, through: :set_member_subjects
   has_many :set_member_subjects, dependent: :destroy
-  has_many :subject_workflow_counts, dependent: :destroy
+  has_many :workflows, through: :set_member_subjects
```

Comment:
yeah - i use the `subject.workflows` method in the remover `workflow_ids = orphan.workflows.map(&:id)`

That could probably change to just a pluck instead too.